### PR TITLE
Update spec templates in response to contributor feedback

### PR DESCRIPTION
**Pull request by jattasNI**
# Pull Request

## 🤨 Rationale

In conversations with contributors we realized there were some unstated requirements for new component development. We should capture these in our spec templates.

The gaps included expectations for:
1. form integration
2. following existing ARIA patterns

## 👩‍💻 Implementation

Updated spec templates for custom components and FAST-based components.

## 🧪 Testing

N/A

## ✅ Checklist

- [X] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: specs/templates/custom-component.md [hunk 1]
```diff
@@ -65,13 +65,19 @@
...
+
 ### Angular integration 
 
 *Describe the plan for Angular support, including directives for attribute binding and ControlValueAccessor for form integration. Depending on the contributor's needs, implementing Angular integration may be deferred but the initial spec should still document what work will be needed.*
 
 ### Blazor integration 
 
-*Describe the plan for Blazor support. See the [nimble-blazor CONTRIBUTING.md](/packages/nimble-blazor/CONTRIBUTING.md) for details. Depending on the contributor's needs, implementing Blazor integration may be deferred but the initial spec should still document what work will be needed.*
+*Describe the plan for Blazor support, including form integration. See the [nimble-blazor CONTRIBUTING.md](/packages/nimble-blazor/CONTRIBUTING.md) for details. Depending on the contributor's needs, implementing Blazor integration may be deferred but the initial spec should still document what work will be needed.*
```
Comment (rajsite) [inline specs/templates/custom-component.md:80]: We don't have great Blazor EditForm support and there is an open issue for that: https://github.com/ni/nimble/issues/766

Maybe remove it from now and add a note to that issue to re-add iit?

File: specs/templates/fast-based-component.md [hunk 2]
```diff
@@ -34,13 +34,18 @@
...
+
+
 ### Angular integration 
 
 *Describe the plan for Angular support, including directives for attribute binding and ControlValueAccessor for form integration. Depending on the contributor's needs, implementing Angular integration may be deferred but the initial spec should still document what work will be needed.*
 
 ### Blazor integration 
 
-*Describe the plan for Blazor support. See the [nimble-blazor CONTRIBUTING.md](/packages/nimble-blazor/CONTRIBUTING.md) for details. Depending on the contributor's needs, implementing Blazor integration may be deferred but the initial spec should still document what work will be needed.*
+*Describe the plan for Blazor support, including form integration. See the [nimble-blazor CONTRIBUTING.md](/packages/nimble-blazor/CONTRIBUTING.md) for details. Depending on the contributor's needs, implementing Blazor integration may be deferred but the initial spec should still document what work will be needed.*
```
Comment (rajsite) [inline specs/templates/fast-based-component.md:48]: same as above

File: specs/templates/custom-component.md [hunk 3]
```diff
@@ -65,13 +65,19 @@
 
 *Work closely with the visual design partner to co-develop the API and anatomy along side the visual design.*
 
+### Native form integration
+
+*Describe the plan for custom element form integration or why it's not necessary.*
+
```
Comment (rajsite) [inline specs/templates/custom-component.md:71]: There should probably be a link in here and discussion about aligning with aria-patterns: https://www.w3.org/WAI/ARIA/apg/patterns/

And falling back to research of aria patterns in the community and  checking open issues in the aria repo

Comment (rajsite) [reply specs/templates/custom-component.md:71]: oh wait, you have that below lol

**Review by rajsite — APPROVED** [review]

File: specs/templates/custom-component.md [hunk 4]
```diff
@@ -65,13 +65,19 @@
```
Comment (rajsite) [inline specs/templates/custom-component.md:65]: @jattasNI reviving this to get it in if we think it's still valuable

File: specs/templates/custom-component.md (hunk 1 above)
Comment (jattasNI) [reply specs/templates/custom-component.md:80]: I started doing that but then noticed that the Blazor CONTRIBUTING does include some basic instructions for form input controls. I updated those instructions to link to the EditForm issue. But I lean towards keeping this addition to remind devs to do those basic steps. Hopefully the link to CONTRIBUTING that's right here will be enough of a breadcrumb.

File: specs/templates/custom-component.md (hunk 4 above)
Comment (jattasNI) [reply specs/templates/custom-component.md:65]: D'oh, I thought I tagged you in this comment when I first put up the PR but accidentally left it in a pending state so it never sent.

> @rajsite The contents of this PR are from my memory of our meeting about the button group plus a few minutes of web searching. I got the sense that you had more in depth requirements in mind. If so, feel free to build on this PR. If you don't have time to do that and you think this is a step in the right direction then we could also submit this and create an issue to track further improvements.

Based on your approval it appears you think it's a step in the right direction so I'll clean it up and complete it. Can you think of anything else we should track in a follow up?

Comment (rajsite) [reply specs/templates/custom-component.md:65]: Think it's good for where we are today. Expecting we'll need to revisit once the spright vision is more ironed out / the what belongs in the nimble library vs spright library idea is clearer

**Review by atmgrifter00 — APPROVED** [review]

**Pull request merged**